src/cls/journal/cls_journal.cc

```diff
@@ -540,10 +540,12 @@ int journal_client_register(cls_method_context_t hctx, bufferlist *in,
                             bufferlist *out) {
   std::string id;
   bufferlist data;
+  std::set<cls::journal::Client> clients;
   try {
     bufferlist::iterator iter = in->begin();
     ::decode(id, iter);
     ::decode(data, iter);
+    ::decode(clients, iter);
```

Comment:
I'd rather see both use a helper method to do the decoding in X record batches.  Not that we should have thousands of clients, but it's a good style to use when working in OSD methods to chunk the work.